### DISCUSSION TITLE
hability to show scroll

### DISCUSSION THREAD

**Discussion by wnleao** [sevdokimov/log-viewer#19] [Q&A]
First of all, great tool! Is there an option to turn on the vertical scroll? Is there a limitation? I believe that implementing an infinite scroll is something that is not trivial.

**Comment by sevdokimov** [comment]
It's impossible to enable vertical scroll because it is not implemented. It may be useful, I'll try to implement it later.

**Reply by wnleao** [reply]
Thank you for your quick response. Would it be okay if I contribute to your project? I could start with the autoscroll and this vertical scroll, for instance. Other things that would be useful are: checkbox to wrap around when searching; button to go to the end of the file.

**Reply by sevdokimov** [reply]
It would be cool! 

> I could start with the autoscroll

Thanks, but I almost implemented that)

> vertical scroll

It is not trivial. No problem when one file is opened: each loaded row contains an offset in the file. But if several files are opened we cannot determine the positions in each file because UI may not contain loaded rows for some files. I propose don't do it now.

> checkbox to wrap around when searching

Could you clarify what do you mean?

> button to go to the end of the file

where do you intend to place this button?

I'll create several tickets in https://github.com/sevdokimov/log-viewer/issues section, you can take one of them

**Reply by wnleao** [reply]
great! I'm going to take a look.

> It is not trivial. No problem when one file is opened: each loaded row contains an offset in the file. But if several files are opened we cannot determine the positions in each file because UI may not contain loaded rows for some files. I propose don't do it now.

we could add the functionality when you have 1 log only. I understand why it would be hard to do it with several log files.

> checkbox to wrap around when searching

usually, most document searches wrap around when you hit the lower or upper end of the file. Like you're searching for something and they don't stop and ask you to use other shortcut to go back on your search. I've some issues, for instance, if you reach the end of a log file when searching for something would it be cheap to start searching at the beginning of the log file?

> where do you intend to place this button?

Not sure where, but on second thought, if you have the hability to autoscroll / lock your console at the end of the file, then this feature may not be that useful

**Reply by sevdokimov** [reply]
> we could add the functionality when you have 1 log only.

Good! It's easy to add scroll for 1 log.

> usually, most document searches wrap around when you hit the lower or upper end of the file. Like you're searching for something and they don't stop and ask you to use other shortcut to go back on your search. I've some issues, for instance, if you reach the end of a log file when searching for something would it be cheap to start searching at the beginning of the log file?

Agree, I saw the situation when a person could not find substring because he reached the end of log and didn't understand that he needs to search back.

> Not sure where, but on second thought, if you have the hability to autoscroll / lock your console at the end of the file, then this feature may not be that useful

A user can press End on the keyboard. There is no good place for this button.

**Reply by wnleao** [reply]
> Good! It's easy to add scroll for 1 log.

I'm going to add an issue then, ok?

> Agree, I saw the situation when a person could not find substring because he reached the end of log and didn't understand that he needs to search back.

Could this be another issue, I'm going to add it aswell.

> A user can press End on the keyboard. There is no good place for this button.

Yeah I agree with you. No need for it.
